lib/spack/spack/test/concretize.py

```diff
@@ -2913,3 +2913,38 @@ def test_spec_filters(specs, include, exclude, expected):
         factory=lambda: specs, is_usable=lambda x: True, include=include, exclude=exclude
     )
     assert f.selected_specs() == expected
+
+
+@pytest.mark.only_clingo("clingo only re-use feature being tested")
+@pytest.mark.regression("38484")
+def test_git_ref_version_can_be_reused(
+    install_mockery_mutable_config, do_not_check_runtimes_on_reuse
+):
+    first_spec = spack.spec.Spec("zlib-ng@git.2.1.5=2.1.5").concretized()
+    first_spec.package.do_install(fake=True, explicit=True)
+
+    with spack.config.override("concretizer:reuse", True):
+        second_spec = spack.spec.Spec("zlib-ng@git.2.1.5=2.1.5~opt").concretized()
+        # is_installed(first_spec)
+        assert second_spec.dag_hash() != first_spec.dag_hash()
```

Comment:
This test would be stronger if you use a non-default variant in `first_spec` and leave the variant un-mentioned in `second_spec`

Reply:
Hmmm that is not a reproducer of the bug reported which is why I didn't do that. In this case I would expect it to reuse.
I can add that to this test though.